modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/util/StreamInterceptorsLoader.java

```diff
@@ -0,0 +1,159 @@
+                OMElement interceptorElem = (OMElement) iterator.next();
+                if (interceptorElem.getAttribute(CLASS_Q) != null) {
+                    String className = interceptorElem.getAttributeValue(CLASS_Q);
+                    if (!"".equals(className)) {
+                        StreamInterceptor interceptor = createInterceptor(className);
+                        interceptors.add(interceptor);
+                        populateParameters(interceptorElem, interceptor);
+                    } else {
+                        handleException("Class name is one or more interceptor");
+                    }
+                } else {
+                    handleException("Class name is one or more interceptor");
+                }
+            }
+        }
+    }
+
+    private static StreamInterceptor createInterceptor(String classFQName) {
+
+        Object obj = null;
+        try {
+            obj = Class.forName(classFQName).newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            handleException("Error creating Interceptor for class name : " + classFQName, e);
+        }
+        if (obj instanceof StreamInterceptor) {
+            return (StreamInterceptor) obj;
+        } else {
+            handleException(
+                    "Error creating StreamInterceptor. The Interceptor should be of type org.apache.synapse.transport.passthru.StreamInterceptor");
+        }
+        return null;
+    }
+
+    private static void populateParameters(OMElement handlerElem, StreamInterceptor interceptor) {
+
+        for (Iterator it = handlerElem.getChildrenWithName(PARAM_Q); it.hasNext(); ) {
+            OMElement child = (OMElement) it.next();
+
+            String propName = child.getAttribute(NAME_ATT).getAttributeValue();
+            if (propName == null) {
+                handleException("StreamInterceptor parameter must contain the name attribute");
+            } else {
+                if (child.getAttribute(VALUE_ATT) != null) {
+                    String value = child.getAttribute(VALUE_ATT).getAttributeValue();
+                    interceptor.addProperty(propName, value);
+                    PropertyHelper.setInstanceProperty(propName, value, interceptor);
+                } else {
+                    OMNode omElt = child.getFirstElement();
+                    if (omElt != null) {
+                        interceptor.addProperty(propName, omElt);
+                        PropertyHelper.setInstanceProperty(propName, omElt, interceptor);
+                    } else {
+                        handleException(
+                                "StreamInterceptor parameter must contain name and value attributes, or a name and a child XML fragment");
+                    }
+                }
+            }
+        }
+    }
+
+    private static void handleException(String msg) {
+        log.error(msg);
+        throw new RuntimeException(msg);
```

Comment:
Is throwing a runtime option a good selection here? The exception will get ignored unless specifically caught, and will cause the server to go to a dead state.

Reply:
yes, server is expected to go to a dead state and to be restarted since this is a config issue